services/aps.js

```diff
@@ -1,75 +1,123 @@
 
-service.getAuthorizationUrl = () => internalAuthClient.generateAuthUrl();
+const service = (module.exports = {});
+
+service.getAuthorizationUrl = () =>
+  authenticationClient.authorize(
+    APS_CLIENT_ID,
+    ResponseType.Code,
+    APS_CALLBACK_URL,
+    new Array(
+      Scopes.Dataread,
+      Scopes.Datacreate,
+      Scopes.Viewablesread
+    )
+  );
 
 service.authCallbackMiddleware = async (req, res, next) => {
-    const internalCredentials = await internalAuthClient.getToken(req.query.code);
-    const publicCredentials = await publicAuthClient.refreshToken(internalCredentials);
-    req.session.public_token = publicCredentials.access_token;
-    req.session.internal_token = internalCredentials.access_token;
-    req.session.refresh_token = publicCredentials.refresh_token;
-    req.session.expires_at = Date.now() + internalCredentials.expires_in * 1000;
-    next();
+  const internalCredentials =
+    await authenticationClient.getThreeLeggedTokenAsync(
+      APS_CLIENT_ID,
+      APS_CLIENT_SECRET,
+      req.query.code,
+      APS_CALLBACK_URL
+    );
+  const publicCredentials = await authenticationClient.getRefreshTokenAsync(
+    APS_CLIENT_ID,
+    APS_CLIENT_SECRET,
+    internalCredentials.refresh_token,
+    new Array(Scopes.Viewablesread)
+  );
+  req.session.public_token = publicCredentials.access_token;
+  req.session.internal_token = internalCredentials.access_token;
+  req.session.refresh_token = publicCredentials.refresh_token;
+  req.session.expires_at = Date.now() + internalCredentials.expires_in * 1000;
+  next();
 };
 
 service.authRefreshMiddleware = async (req, res, next) => {
-    const { refresh_token, expires_at } = req.session;
-    if (!refresh_token) {
-        res.status(401).end();
-        return;
-    }
+  const { refresh_token, expires_at } = req.session;
+  if (!refresh_token) {
+    res.status(401).end();
+    return;
+  }
 
-    if (expires_at < Date.now()) {
-        const internalCredentials = await internalAuthClient.refreshToken({ refresh_token });
-        const publicCredentials = await publicAuthClient.refreshToken(internalCredentials);
-        req.session.public_token = publicCredentials.access_token;
-        req.session.internal_token = internalCredentials.access_token;
-        req.session.refresh_token = publicCredentials.refresh_token;
-        req.session.expires_at = Date.now() + internalCredentials.expires_in * 1000;
-    }
-    req.internalOAuthToken = {
-        access_token: req.session.internal_token,
-        expires_in: Math.round((req.session.expires_at - Date.now()) / 1000)
-    };
-    req.publicOAuthToken = {
-        access_token: req.session.public_token,
-        expires_in: Math.round((req.session.expires_at - Date.now()) / 1000)
-    };
-    next();
+  if (expires_at < Date.now()) {
+    const internalCredentials = await authenticationClient.getRefreshTokenAsync(
+      APS_CLIENT_ID,
+      APS_CLIENT_SECRET,
+      {
+        refresh_token,
```

Comment:
Is this parameter really an object? According to the method signature this should just be a string.